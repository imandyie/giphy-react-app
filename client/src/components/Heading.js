import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
import '../styles/components/Heading.scss';
import colors from '../styles/colors';

export default class Heading extends PureComponent {
  render() {
    const {
      text,
      color,
      isFetching
    } = this.props;
    const fetchingClass = isFetching ? 'animated-heading-placeholder' : '';

    return (
      <h1
        className={`heading ${fetchingClass}`}
        style={{color}}
      >
        {this.props.text}
      </h1>
    );
  }
}

Heading.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string,
  isFetching: PropTypes.bool
};