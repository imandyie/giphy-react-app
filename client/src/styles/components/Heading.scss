@import '../mixins/responsive.scss';
@import '../colors.scss';

.heading {
  margin: 0;
  padding: 0;
  font-weight: lighter;
  font-size: 2.4rem;
}

@keyframes headingPlaceholder {
  0% {
    background-position: -400px 0
  }
  100% {
    background-position: 400px 0
  }
}

.animated-heading-placeholder {
  animation-duration: 0.7s;
  animation-fill-mode: forwards;
  animation-iteration-count: infinite;
  animation-name: headingPlaceholder;
  animation-timing-function: linear;
  background: $white;
  background: linear-gradient(to right, $darkerGray 8%, $darkGray 18%, $darkerGray 33%);
  background-size: 800px 104px;
  height: 50px;
  margin-bottom: 5px;
  position: relative;
  width: 400px;
  color: transparent !important;
}

/** Responsive **/
@include responsive(sm) {
  .heading {
    font-size: 2rem;
  }

  .animated-heading-placeholder {
    width: 100%;
  }
}