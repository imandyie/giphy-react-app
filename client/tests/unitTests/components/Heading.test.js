import React from 'react';
import Heading from '../../../src/components/Heading';
import renderer from 'react-test-renderer';

test('Heading renders without errors', () => {
  const component = renderer.create(
    <Heading
      text="Sample Heading"
      color="red"
      isFetching={true}
    />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
