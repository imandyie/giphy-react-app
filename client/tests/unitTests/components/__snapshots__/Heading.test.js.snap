// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders without errors 1`] = `
<h1
  className="heading animated-heading-placeholder"
  style={
    Object {
      "color": "red",
    }
  }
>
  Sample Heading
</h1>
`;
